website/theme/components/Announcement/index.tsx

```diff
@@ -1,49 +1,43 @@
 import { useState } from 'react';
-import { usePageData } from 'rspress/runtime';
-import { useLang } from 'rspress/runtime';
-import { useI18n } from '../../i18n';
 import IconCloseCircle from './close';
+import styles from './index.module.scss';
 
-export function Announcement() {
-  const t = useI18n();
+export function Announcement({
+  href,
+  message,
+  // we don't need localStorageKey for persist to recommend upgrading
+  // localStorageKey,
+  display = true,
+}: {
+  href: string;
+  message: string;
+  // localStorageKey: string;
+  display?: boolean;
+}) {
+  if (!display) {
+    return null;
+  }
   const [disable, setDisable] = useState(
-    window.localStorage.getItem('disabled-hire') ?? false,
+    // window.localStorage.getItem(localStorageKey) ?? false,
+    false,
   );
-  const { page } = usePageData();
-  const lang = useLang();
-  // Only display in homepage
-  if (page.pageType !== 'home' || disable) {
+
+  if (disable) {
     return null;
   }
+
   return (
-    <div
-      className="h-8 flex justify-center items-center bg-gradient-to-r from-green-400 via-yellow-300 to-orange-500"
-      style={{
-        height: '2rem',
-        display: 'flex',
-        justifyContent: 'center',
-        alignItems: 'center',
-      }}
-    >
-      <a
-        href={`${lang == 'en' ? '' : `/${lang}`}/misc/join-us.html`}
-        className="hover:underline text-gray-700 font-bold"
-      >
-        {t('recruit')}
+    <div className={styles.bar}>
+      <a href={href} className={styles.link}>
+        {message}
       </a>
       <IconCloseCircle
         onClick={() => {
           setDisable(true);
-          window.localStorage.setItem('disabled-hire', 'true');
-        }}
-        style={{
-          right: 10,
-          color: 'white',
-          fontSize: 18,
-          position: 'absolute',
-          cursor: 'pointer',
+          // window.localStorage.setItem(localStorageKey, 'true');
```

Comment:
The close icon can be mored if we do not use local storage

Reply:
done